resources/aws/securitygroup.go

```diff
@@ -35,27 +35,17 @@ func (s SecurityGroup) findExisting() (*ec2.SecurityGroup, error) {
 			},
 		},
 	})
-	if err != nil {
-		return nil, microerror.MaskAny(err)
-	}
-
-	if len(securityGroups.SecurityGroups) < 1 {
-		return nil, microerror.MaskAny(securityGroupFindError)
-	}
-
-	return securityGroups.SecurityGroups[0], nil
+	return out.SecurityGroups, microerror.MaskAny(err)
 }
 
 func (s *SecurityGroup) CreateIfNotExists() (bool, error) {
 	if err := s.CreateOrFail(); err != nil {
 		if strings.Contains(err.Error(), awsclient.SecurityGroupDuplicate) {
-			securityGroup, err := s.findExisting()
-			if err != nil {
-				return false, microerror.MaskAny(err)
+			securityGroups, err := s.list()
+			if err == nil && len(securityGroups) > 0 {
+				s.id = *securityGroups[0].GroupId
 			}
-			s.id = *securityGroup.GroupId
-
-			return false, nil
+			return false, microerror.MaskAny(err)
```

Comment:
I'm not sure if this makes the intent clearer. Yes, `microerror.MaskAny(nil)` returns `nil`, but I find the previous implementation more explicit.